src/pages/SitePermissionPage.qml

```diff
@@ -0,0 +1,94 @@
+/****************************************************************************
+**
+** Copyright (c) 2020 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.6
+import Sailfish.Silica 1.0
+import Sailfish.WebView.Controls 1.0
+
+Page {
+    id: page
+
+    property string title
+    property string url
+
+    property var _permissions: {
+        "geolocation": PermissionManager.Prompt
+    }
+
+    signal requestFinished()
+
+    function getCurrentIndex(type) {
+        switch(_permissions[type]) {
+        case PermissionManager.Allow:
+            return 0 // index for "Allow" menu
+        case PermissionManager.Deny:
+            return 1 // index for "Deny" menu
+        default:
+            return 2 // index for "Always ask" menu
+        }
+    }
+
+    ListModel {
+        id: permissionTypesModel
+        ListElement {
+            //% "Location"
+            title: qsTrId("sailfish_browser-ti-location")
+            type: "geolocation"
+        }
+        Component.onCompleted: {
+            PermissionManager.requestUriPermissions(url, function(perms) {
+                for (var i in perms) {
+                    var type = perms[i].type
+                    _permissions[type] = parseInt(perms[i].capability)
+                }
+                requestFinished()
+            })
+        }
+    }
+
+    SilicaListView {
+        id: view
+        anchors.fill: parent
+        header: PageHeader {
+            title: page.title
+            description: page.url
+        }
+
+        model: permissionTypesModel
+
+        delegate: ComboBox {
+            label: model.title
+
+            currentIndex: getCurrentIndex(model.type)
```

Comment:
It's not clear to me what this line is doing. Won't this get immediately overwritten by line 72?